packages/realm/src/property-accessors/Array.ts

```diff
@@ -87,11 +87,12 @@ export function createArrayPropertyAccessor({
         assert.inTransaction(realm);
         assert.iterable(values);
 
+        const valuesCopiedRef = Array.from(values);
```

Comment:
Using `Array.from` will eagerly consume the `values` iterator, which might be okay, but depending on the size of the collection, it could cause unnecessary memory pressure. I suggest using a snapshot instead.